test/e2e/networkingnew.go

```diff
@@ -0,0 +1,358 @@
+		expectNoError(err)
+		defer f.Client.Pods(f.Namespace.Name).Delete(podName, nil)
+
+		By("Verify that the pod succeed")
+		expectNoError(waitForPodSuccessInNamespace(f.Client, podName, contName, f.Namespace.Name))
+	})
+
+	// First test because it has no dependencies on variables created later on.
+	It("should provide unchanging, static URL paths for kubernetes api services.", func() {
+		tests := []struct {
+			path string
+		}{
+			{path: "/validate"},
+			{path: "/healthz"},
+			// TODO: test proxy links here
+		}
+		for _, test := range tests {
+			By(fmt.Sprintf("testing: %s", test.path))
+			data, err := f.Client.RESTClient.Get().
+				Namespace(f.Namespace.Name).
+				AbsPath(test.path).
+				DoRaw()
+			if err != nil {
+				Failf("Failed: %v\nBody: %s", err, string(data))
+			}
+		}
+	})
+
+	//Each tuple defined in this struct array represents
+	//a number of services, and a timeout.  So for example,
+	//{1, 300} defines a test where 1 service is created and
+	//we give it 300 seconds to timeout.  This is how we test
+	//services in parallel... we can create a tuple like {5,400}
+	//to confirm that services over 5 ports all pass the networking test.
+	serviceSoakTests := []struct {
+		service        int
+		timeoutSeconds time.Duration
+	}{
+		//These are very liberal, once this test is running regularly,
+		//We will DECREASE the timeout value.
+		//Replace this with --scale constants eventually https://github.com/kubernetes/kubernetes/issues/10479.
+		{1, time.Duration(100 * time.Second)},
+		{3, time.Duration(200 * time.Second)}, //Test that parallel nettests running on different ports complete.
+	}
+
+	for _, svcSoak := range serviceSoakTests {
+		//copy to local to avoid range overwriting
+		timeoutSeconds := svcSoak.timeoutSeconds
+		serviceNum := svcSoak.service
+		It(fmt.Sprintf("should function for intrapod communication between all hosts in %v parallel services", serviceNum),
+			func() {
+				Logf("running service test with timeout = %v for %v", timeout, serviceNum)
+				runNetTest(timeoutSeconds, f, makePorts(serviceNum), nettestVersion)
+			})
+	}
+})
+
+//pollPeerStatus will either fail, pass, or continue polling.
+//When completed, it will write the service name to the channel provided, thus
+//facilitating parallel service testing.
+func pollPeerStatus(serviceDoneChannel chan string, f *Framework, svc *api.Service, pollTimeoutSeconds time.Duration) {
+
+	Logf("Begin polling " + svc.Name)
+	getDetails := func() ([]byte, error) {
+		return f.Client.Get().
+			Namespace(f.Namespace.Name).
+			Prefix("proxy").
+			Resource("services").
+			Name(svc.Name).
+			Suffix("read").
+			DoRaw()
+	}
+
+	getStatus := func() ([]byte, error) {
+		return f.Client.Get().
+			Namespace(f.Namespace.Name).
+			Prefix("proxy").
+			Resource("services").
+			Name(svc.Name).
+			Suffix("status").
+			DoRaw()
+	}
+
+	passed := false
+
+	expectNoError(wait.Poll(2*time.Second, pollTimeoutSeconds, func() (bool, error) {
```

Comment:
`wait.Poll` currently guarantees you'll have a 2-second delay before your condition func is called. If you would prefer to check without the delay, use `wait.PollImmediate`.